zlib-ng.h.in

```diff
@@ -130,11 +130,11 @@ typedef struct zng_gz_header_s {
     int32_t         xflags;     /* extra flags (not used when writing a gzip file) */
     int32_t         os;         /* operating system */
     uint8_t         *extra;     /* pointer to extra field or NULL if none */
+    uint8_t         *name;      /* pointer to zero-terminated file name or NULL */
+    uint8_t         *comment;   /* pointer to zero-terminated comment or NULL */
     uint32_t        extra_len;  /* extra field length (valid if extra != NULL) */
     uint32_t        extra_max;  /* space at extra (only when reading header) */
-    uint8_t         *name;      /* pointer to zero-terminated file name or NULL */
     uint32_t        name_max;   /* space at name (only when reading header) */
-    uint8_t         *comment;   /* pointer to zero-terminated comment or NULL */
     uint32_t        comm_max;   /* space at comment (only when reading header) */
     int32_t         hcrc;       /* true if there was or will be a header crc */
     int32_t         done;       /* true when done reading gzip header (not used when writing a gzip file) */
```

Comment:
I think here `name_max` did mess the alignment as it is only 4 bytes (there is no `name_len` or `comm_len` members), but `comment` needs 8-byte alignment. After the change `name_max` and `comm_max` fill 8-byte "slot" together and `hcrc` and `done` the next 8-byte "slot".